### DISCUSSION TITLE
Contributing to Koka?

### DISCUSSION THREAD

**Discussion by chtenb** [koka-lang/koka#381] [General]
Hi everyone

First of all, I think Koka is a really cool and interesting project! Thanks for creating it.

However, while using it I'm stumbling over issues/bugs that would need resolving in order to use Koka for real projects (but still non-production).
I would not mind digging into it and contributing fixes.
But looking at the PR track record, it seems that PRs don't often make it into the main branch. (This is not a critique, just an observation.)

1. Does real-world usage align with the goals of the Koka project? I.e. if the goal is strictly to be a research language used for reference implementations of new technologies, which would be totally understandable, then that probably does not align with real-world usage.
2. Is Koka open to contributions? If yes, what is the preferred path for making a contribution?

Of course "Not yet" is also a valid answer to both questions, and would make it clear to potential contributors to give it a rest until the core language has matured more.

**Comment by TimWhiting** [comment]
As an official contributor, as well as PhD researcher, here are my observations.

1. Real world usage is in a not-yet but soon category. I have no doubt that research will still be a large part of Koka for the near future. That being said research is only as good as it is applicable to real world code.

2. The preferred path is to create PRs. Daan has other important projects he works on at Microsoft, so he doesn't always have a ton of time for Koka, but he is interested in contributions. It would be great to have more people that he trusts to contribute quality PRs. Fixes are more likely to be accepted quickly than additions. My PRs sat for awhile before a large majority of them being merged today. See the dev branch for the latest changes and fixes. Unfortunately there might be a few regressions as well on the dev branch since Daan is in the middle of overhauling the inference algorithm to have better performance.
